DEPARTMENTAL TRAINEES/ASSISTANTS!

DrDiasyl opened the pull request:
**TGstation discord thread where you can leave any feedback during those test merges:**
https://discord.com/channels/326822144233439242/1092828538912116806 
## About The Pull Request
Adds into game departmental assistants/trainees, those are junior roles of their respective departments.
They answer to everyone in their department, designed for new players (only Security Cadet has a time requirement).
![image](https://user-images.githubusercontent.com/42353186/230651843-f497fb5e-48c5-4b49-8b64-23dca51aebbd.png)
**From left to right:**
- Test Subject
- Nurse
- Security Cadet
- Maintenance Technician

Now to get into any department, you need to play at least 1 round (60 minutes) on the departmental trainee to get access to the entire department. This won't affect any old/pro players as they already have enough minutes in their department to have it fully unlocked.

- Medical still has Paramedic accessible without playing 60 minutes on Nurse, but to become Medical Doctor you need 60 minutes on either Nurse or Paramedic.
- To become Security Officer you need to first play 120 minutes on the crew to get onto Cadet, and then 180 minutes on the Cadet to become Officer. (Prior that you needed 300 minutes on crew to play Security Officer, I just divided it between those)
- To become Engineer/Atmos Technician you need to play 60 minutes on the Maintenance Technician.

All of the departmental assistants spawn with less round start gear and lesser access to prevent possible problems with that of tiders evolving.
Only Engineering, Security, Science, and Medical got departmental trainees as they are the most difficult to learn. Departments like Service and Cargo are simple enough to have people join in any of their roles straightaway!

Also have added jumpskirt variations for some jumpsuits that lacked them.
## Why It's Good For The Game

1. It unlocks additional job slots in the game, now stations can fit more people inside them.
2. Helps newbies learn new roles by being able to join a department, without having harsh requirements but with still sufficient access to have a good ground for learning.
3. Decrease the number of shitsecs by making the learning curve for Security more logical. Now instead of spending 300 hours on the crew and joining sec with almost no idea what to do, you must spend 120 minutes on the crew to understand relatively who they are, and then 180 minutes on the Cadet to learn how to do their job properly. (For anyone with F on Algebra, 180+120=300 so in general time didn't change)

You might say _"Just hire an assistant on role lol"_, but I will say that there are only 2 types of grey shirts:
1. People who can't talk, walk, or even read
2. Meta-power gamers who join Assistant just for full freedom and commit GREYTIDE!

As such assistants never make a good job supplement, we are better off with assistants being as super newbie jobs to learn walking, talking, etc. While departmental assistants can work as a role to learn certain departments.
## Changelog
:cl:
add: Added departmental trainees! These are the most junior roles in the Engineering (Maintenance Technician), Security (Security Cadet), Science (Test Subject), and Medical (Nurse) departments to help newbies learn those roles without taking stressful jobs full of responsibilities!
add: Security time requirement is now divided between Security Cadet and Security Officers. 300 minutes on the crew to become Security Officer, have turned into 120 minutes on the crew to become Cadet, and then 180 minutes to become a Security Officer in an effort to reduce shitcurity.
imageadd: Now Grey Security jumpsuit has a jumpskirt variation!
imageadd: Now Engineering Hazard jumpsuit has a jumpskirt variation!
imageadd: Now the Nurse uniform is divided into 2 - the new one which is a Medical Doctor jumpsuit but with a pink cross which also has a jumpskirt variation, and the old costume-like one.
/:cl:

DrDiasyl committed:
adds some assistants

defines all of the dep assistant jobs.
gets them all a trim and hud sprite.

gives adjutant his own suit and plasma men clothes.

DrDiasyl committed:
adds more roles, add some clothes

Now grey security jumpsuits and engineering hazard jumpsuits have a skirt variation.

Added the rest of the departmental assistants.

Now to play most of the department's main roles, you need to spend at least 60 minutes on a departmental assistant of that department.

DrDiasyl committed:
fixes steward icon missing, ticks science student

Now also cargo clerk got his own uniform.
Security cadet tie now spawns tied.
Some dep assistants are restricted from certain traitor missions.
Now maint tech doesn't spawn with two PDAs.

DrDiasyl committed:
now science student is a test subject

Also a little text formatting of job descriptions

DrDiasyl committed:
removes science student mentions

DrDiasyl committed:
Adds icons for the jobs and etc

The nurse now has a normal uniform version.
Formating fixes

DrDiasyl committed:
fixes fixes

DrDiasyl committed:
Deletes steward, cargo clerk, and adjutant

Steward and Cargo Clerk are pretty useless as Cargo and Service is pretty easy jobs anyway.

The Adjutant will come back in a separate PR.

DrDiasyl committed:
Adds in diffuclty and assigns, does trim cleanup

DrDiasyl committed:
Revert "Adds in diffuclty and assigns, does trim cleanup"

This reverts commit 5e00a6b9978b5d0f24bae21646197c4868a34a89.

DrDiasyl committed:
Reduces dep assitants slots from 5 to 3 on every

Also some cleanup

DrDiasyl committed:
grammar and returns test subject jumpsuit

DrDiasyl committed:
Black sneakers more fashionable for test subject

DrDiasyl committed:
Makes cadet access lesser

Now he can't reach seccies lockers, open perma or any other prison cells

DrDiasyl committed:
Merge remote-tracking branch 'upstream/master' into departmental-assistants/newbies

DrDiasyl committed:
Merge remote-tracking branch 'upstream/master' into departmental-assistants/newbies

DrDiasyl committed:
Changes time req for cadet and officer

Also hud icons update to make them clearer

flowercuco commented on code/modules/clothing/under/jobs/command.dm:
```diff
@@ -46,3 +47,24 @@
 	can_adjust = FALSE
+
+//ADJUTANT
+/obj/item/clothing/under/rank/adjutant
+	desc = "It's a light blue suit, its resistant to the hardest profanities and coffee spills."
+	name = "adjutant's suit"
+	icon_state = "adjutantsuit"
+	inhand_icon_state = "b_suit"
+	sensor_mode = SENSOR_COORDS
+	random_sensor = FALSE
+	icon = 'icons/obj/clothing/under/suits.dmi'
+	worn_icon = 'icons/mob/clothing/under/suits.dmi'
+	armor_type = /datum/armor/rank_captain
+
+/obj/item/clothing/under/rank/adjutant/skirt
+	desc = "It's a light blue suitskirt, its resistant to the hardest profanities and coffee spills."
+	name = "adjutant's suitskirt"
+	icon_state = "adjutantsuit_skirt"
+	body_parts_covered = CHEST|GROIN|ARMS
+	dying_key = DYE_REGISTRY_JUMPSKIRT
+	female_sprite_flags = FEMALE_UNIFORM_TOP_ONLY
+	supports_variations_flags = CLOTHING_DIGITIGRADE_VARIATION_NO_NEW_ICON
```
accidentally left some adjutant stuff in here

flowercuco commented on code/modules/clothing/under/jobs/Plasmaman/command.dm:
```diff
@@ -1,3 +1,11 @@
+/obj/item/clothing/under/plasmaman/adjutant
+	name = "adjutant's plasma envirosuit"
+	desc = "It's a light blue envirosuit, its resistant to the hardest profanities and coffee spills."
+	icon_state = "adjutant_envirosuit"
+	inhand_icon_state = null
+	sensor_mode = SENSOR_COORDS
+	random_sensor = FALSE
+	armor_type = /datum/armor/plasmaman_captain
```
here too

DrDiasyl replied:
fuck, I also forgot to move the adjutant plasmamen suit!

Fikou commented on code/modules/clothing/under/jobs/engineering.dm:
```diff
@@ -62,7 +62,14 @@
 	inhand_icon_state = "syndicate-orange"
 	alt_covers_chest = TRUE
-
+/obj/item/clothing/under/rank/engineering/engineer/hazard/skirt
```
new paths must be spaced out with a newline

Fikou commented on code/modules/clothing/under/jobs/medical.dm:
```diff
@@ -133,6 +133,20 @@
 	supports_variations_flags = CLOTHING_DIGITIGRADE_VARIATION_NO_NEW_ICON
 
 /obj/item/clothing/under/rank/medical/doctor/nurse
```
again here, dont group it all up together its bad to read

Fikou commented on code/modules/jobs/job_types/maintenance_technician.dm:
```diff
@@ -0,0 +1,56 @@
+	bounty_types = CIV_JOB_ENG
+	departments_list = list(
+		/datum/job_department/engineering,
+		)
```
```suggestion
	)
```

Fikou commented on code/modules/jobs/job_types/maintenance_technician.dm:
```diff
@@ -0,0 +1,56 @@
+		/obj/item/lightreplacer = 10,
+		/obj/item/holosign_creator/engineering = 8,
+		/obj/item/wrench/bolter = 8,
+		/obj/item/clothing/head/utility/hardhat/red/upgraded = 1
```
```suggestion
		/obj/item/clothing/head/utility/hardhat/red/upgraded = 1,
```

DrDiasyl replied:
I literally copy pasted engineer code, it is the same there lmao

Fikou replied:
its bad there too

Fikou commented on code/modules/jobs/job_types/security_cadet.dm:
```diff
@@ -0,0 +1,67 @@
+		/obj/item/food/donut/matcha = 10,
+		/obj/item/food/donut/blumpkin = 5,
+		/obj/item/clothing/mask/whistle = 5,
+		/obj/item/melee/baton/security/boomerang/loaded = 1
```
```suggestion
		/obj/item/melee/baton/security/boomerang/loaded = 1,
```

Fikou commented on code/modules/jobs/job_types/security_cadet.dm:
```diff
@@ -0,0 +1,67 @@
+	bounty_types = CIV_JOB_SEC
+	departments_list = list(
+		/datum/job_department/security,
+		)
```
```suggestion
	)
```

Fikou commented on code/modules/jobs/job_types/security_cadet.dm:
```diff
@@ -0,0 +1,67 @@
+	uniform = /obj/item/clothing/under/rank/security/officer/grey
+	backpack_contents = list(
+		/obj/item/evidencebag = 1,
+		)
```
```suggestion
	)
```

Fikou commented on code/datums/id_trim/jobs.dm:
```diff
@@ -1084,7 +1165,32 @@
 		)
 	job = /datum/job/station_engineer
-
+/datum/id_trim/job/test_subject
```
again

DrDiasyl committed:
Cleanup leftovers of adjutant code

Yeah, I also wanted to add in a command assistant.

Adjutant will come out soonTM (really, I'm not joking)

Fikou commented on code/modules/clothing/under/jobs/cargo.dm:
```diff
@@ -36,6 +36,24 @@
 	dying_key = DYE_REGISTRY_JUMPSKIRT
 	female_sprite_flags = FEMALE_UNIFORM_TOP_ONLY
 
+/obj/item/clothing/under/rank/cargo/clerk
```
unused code

Fikou commented on code/modules/clothing/under/jobs/rnd.dm:
```diff
@@ -122,3 +122,19 @@
 	supports_variations_flags = CLOTHING_DIGITIGRADE_VARIATION_NO_NEW_ICON
+
+/obj/item/clothing/under/rank/rnd/test_subject
+	desc = "That jumpsuit you're wearing looks stupid. On other people it looks fine, but scientist has noted that on you it looks 'stupid.'"
```
a/the scientist or scientists

Fikou commented on code/modules/jobs/job_types/security_cadet.dm:
```diff
@@ -0,0 +1,67 @@
+	total_positions = 3
+	spawn_positions = 3
+	supervisors = "the security department and Head of Security"
+	minimal_player_age = 7
```
this means you need to have played 7 days, you sure you want that for this?

DrDiasyl replied:
No, I forgot to delete this

Fikou commented on code/modules/jobs/job_types/test_subject.dm:
```diff
@@ -0,0 +1,51 @@
+	mail_goodies = list(
+		/obj/item/raw_anomaly_core/random = 10,
+		/obj/item/disk/design_disk/bepis = 2,
+		/obj/item/camera_bug = 1
```
```suggestion
		/obj/item/camera_bug = 1,
```

Fikou commented on code/modules/jobs/job_types/test_subject.dm:
```diff
@@ -0,0 +1,51 @@
+	bounty_types = CIV_JOB_SCI
+	departments_list = list(
+		/datum/job_department/science,
+		)
```
```suggestion
	)
```

Fikou commented on code/modules/jobs/job_types/nurse.dm:
```diff
@@ -0,0 +1,59 @@
+	bounty_types = CIV_JOB_MED
+	departments_list = list(
+		/datum/job_department/medical,
+		)
```
```suggestion
	)
```

Fikou commented on code/datums/id_trim/jobs.dm:
```diff
@@ -626,6 +626,35 @@
+		ACCESS_MECH_ENGINE,
+		ACCESS_MINERAL_STOREROOM,
+		ACCESS_TECH_STORAGE,
+		)
```
unindent all these in the file

DrDiasyl replied:
Alright, but you owe my GBP for code improvement then haha

Fikou replied:
youll already get it from sprites but sure

DrDiasyl committed:
Cleaned up the rest of adjutant, also readability and line adjustment

Fikou reviewed (changes requested):
overall im interested in at least trying them out

Mothblocks commented:
I don't think I like this at all.

1. I think incompetent crew members in important positions is good for the game and creates more interesting stories than every doctor/engineer knowing how to do everything. Of course after one round on this you won't be a master, but I think the implied values of this PR are misguided. For example, I completely disagree that "Decrease the number of shitsecs by making the learning curve for Security more logical" is a good thing--you and I do not see the game in the same way.
2. If you're not giving them the full gear and access, you're not really teaching them the role, since when they get the "full" role, they're still going to have another learning curve. People in these roles are going to have to do more that the role they actually want to play does.
3. Opening more job slots shouldn't be treated as an innate good. We have them for a reason and increasing them should be done carefully. Something like increasing job slots of security (assuming you haven't matched the security config, which you might have) for instance is definitely a pretty large change.
4. It makes the jobs menu more cluttered with jobs experienced players shouldn't actually really be playing.

I think what I would want more is the ability for intern stuff to apply to specific jobs. For instance, less than 2 hours prefixes "Junior" to your ID card.

Fikou wants a test merge and I am okay with that but I don't want this merged

DrDiasyl committed:
Give me my GBP for code cleanup

De indented stuff

ChungusGamer666 commented:
When the roles are bloated!

DrDiasyl committed:
Whoops deteled wrong cargo clothes

DrDiasyl commented:
> I don't think I like this at all.
> 
> 1. I think incompetent crew members in important positions is good for the game and creates more interesting stories than every doctor/engineer knowing how to do everything. Of course after one round on this you won't be a master, but I think the implied values of this PR are misguided. For example, I completely disagree that "Decrease the number of shitsecs by making the learning curve for Security more logical" is a good thing--you and I do not see the game in the same way.
> 2. If you're not giving them the full gear and access, you're not really teaching them the role, since when they get the "full" role, they're still going to have another, different learning curve. People in these roles are going to have to do more that the role they actually want to play does.
> 3. Opening more job slots shouldn't be treated as an innate good. We have them for a reason and increasing them should be done carefully. Something like increasing job slots of security (assuming you haven't matched the security config, which you might have) for instance is definitely a pretty large change.
> 4. It makes the jobs menu more cluttered with jobs experienced players shouldn't actually really be playing.
> 
> I think what I would want more is the ability for intern stuff to apply to specific jobs. For instance, less than 2 hours as the job slot prefixes "Junior" to your ID card.
> 
> Fikou wants a test merge and I am okay with that but I don't want this merged

1. I have no idea how you enjoy having shitcurity
2. They don't get cut off really badly, Maint Tech loses almost no access, Nurses losses only Pharmacy access, and the Security Cadet has access to the Security Officer lockers! If they are actually newbies and eager to learn those things, they can be given access to them. I did this mainly so heads can look and see if this is a power shitter or an actual newbie, to grant him access.
3. I heard something-something about server expansion
4. 4 jobs won't cause people's eyes to bleed out.

the thing with prefixes like TGMC rank is kinda interesting, but not the same. You still take up a job slot and have full responsibilities (bonus points if you're the only person in the department), maybe some heads will notice your prefix and won't go harsh _but I think is not that likely._

Likteer commented:
I like the test subjects.

DrDiasyl commented:
> > I don't think I like this at all.
> > 
> > 1. I think incompetent crew members in important positions is good for the game and creates more interesting stories than every doctor/engineer knowing how to do everything. Of course after one round on this you won't be a master, but I think the implied values of this PR are misguided. For example, I completely disagree that "Decrease the number of shitsecs by making the learning curve for Security more logical" is a good thing--you and I do not see the game in the same way.
> > 2. If you're not giving them the full gear and access, you're not really teaching them the role, since when they get the "full" role, they're still going to have another, different learning curve. People in these roles are going to have to do more that the role they actually want to play does.
> > 3. Opening more job slots shouldn't be treated as an innate good. We have them for a reason and increasing them should be done carefully. Something like increasing job slots of security (assuming you haven't matched the security config, which you might have) for instance is definitely a pretty large change.
> > 4. It makes the jobs menu more cluttered with jobs experienced players shouldn't actually really be playing.
> > 
> > I think what I would want more is the ability for intern stuff to apply to specific jobs. For instance, less than 2 hours as the job slot prefixes "Junior" to your ID card.
> > Fikou wants a test merge and I am okay with that but I don't want this merged
> 
> 1. I have no idea how you enjoy having shitcurity
> 2. They don't get cut off really badly, Maint Tech loses almost no access, Nurses losses only Pharmacy access, and the Security Cadet has **NO** access to the Security Officer lockers! If they are actually newbies and eager to learn those things, they can be given access to them. I did this mainly so heads can look and see if this is a power shitter or an actual newbie, to grant him access.
> 3. I heard something-something about server expansion
> 4. 4 jobs won't cause people's eyes to bleed out.
> 
> the thing with prefixes like TGMC rank is kinda interesting, but not the same. You still take up a job slot and have full responsibilities (bonus points if you're the only person in the department), maybe some heads will notice your prefix and won't go harsh _but I think is not that likely._

DrDiasyl commented:
Oops, I meant that Cadet has **NO** access to the security lockers

tralezab commented:
IMO, no, but "test subject" would be an interesting station trait job if it spawned as monkey ala goon

DrDiasyl commented:
> IMO, no, but "test subject" would be an interesting station trait job if it spawned as monkey ala goon

But why _no?_ Only Mothblocks explained what he dislikes.

DrDiasyl committed:
Die adjutant stuff, die

Rex9001 commented:
> IMO, no, but "test subject" would be an interesting station trait job if it spawned as monkey ala goon

Could work, should most likely be held off on until the cargorilla is improved (iirc someone was working on it)

anconfuzedrock commented:
my experience with trainee jobs on other servers has always been problematic.  At best, they're regulated OOC (which TG wouldn't do, which isn't a bad thing) to ensure the role is only played by people wanting to learn the job, and that can't be antag to promote people acting in good faith to teach trainees.

At worst, Trainee is instead played by longtime players who just want to play a certain job with less responsibility. kind of like assistant. I can guarantee you people will play "nurse" due to the name alone.

DrDiasyl committed:
Merge branch 'master' into departmental-assistants/newbies

DrDiasyl commented:
> my experience with trainee jobs on other servers has always been problematic. At best, they're regulated OOC (which TG wouldn't do, which isn't a bad thing) to ensure the role is only played by people wanting to learn the job, and that can't be antag to promote people acting in good faith to teach trainees.
> 
> At worst, Trainee is instead played by longtime players who just want to play a certain job with less responsibility. kind of like assistant. I can guarantee you people will play "nurse" due to the name alone.

Nothing really bad with people playing roles to assist other people with fewer responsibilities, it will only turn bad if those people join in not to assist but to powergame

Mothblocks commented:
> I have no idea how you enjoy having shitcurity

Because conflict is good, people acting on incomplete information is good, and people having wildly different skill levels means interacting with them becomes significantly different.

> I heard something-something about server expansion

That might be true (I'm not in Discord), and if it is the we shouldn't be peppering over problems with job slots by adding more jobs that are separate from the core, but effectively the same.

> 4 jobs won't cause people's eyes to bleed out.

We already have a lot of jobs and any addition to this already massive list needs to have a very good justification and I very much don't agree with yours

DrDiasyl commented:
> Member

Alright, sure whatever, as it seems this won't be merged for _an indefinite_ amount of time I'm happy with at least seeing how good it will be on test merge.
If it will be good on test merge is there _any chance_ this can get real?

Mothblocks commented:
I don't think so, I don't think there's any short term data point that could disagree with my issues. Fikou wants a TM and I respect them so I'm giving them that opportunity

Loafin34 commented:
I want steward and Test Subject fully merged, otherwise Mothblocks has good points. I love the idea of a service assistant that helps the bartender/chef/botanist/clown out with certain tasks.

Bawhoppen commented:
While I appreciate the effort you put into this PR, this is one of those features that if we wanted for the game, we would've added it a very long time ago. It just doesn't really fit that well.

SomeoneYouProbablyKnow commented:
Why are we acting like shitcurity is an issue with the learning curve? I don't think i've seen shisec and though "oh they just don't know how to play" it's a player mindset and culture issue.

Incomeptent sec is not the same as shitsec. Because if we start acting like not being able to do your job perfectly is bad then suddenly shotgunmaxxing or whatever shit sec mains are doing becomes how you're supposed to play the game and i do not want that

DrDiasyl commented:
> Why are we acting like shitcurity is an issue with the learning curve? I don't think i've seen shisec and though "oh they just don't know how to play" it's a player mindset and culture issue.
> 
> Incomeptent sec is not the same as shitsec. Because if we start acting like not being able to do your job perfectly is bad then suddenly shotgunmaxxing or whatever shit sec mains are doing becomes how you're supposed to play the game and i do not want that

This sounds kinda confusing.
Just if the person spends some time learning the job on a lower-competency job first, then there will be lower chances he will try to abuse his powers. Lower but not absolute chance.

moocowswag commented:
Im worried itl make the job screen alittle too cluttered and complex for brand new players and the names for some of the jobs dont do much to help (mainly test subject)

TLDR: Assistant inflation.

SomeoneYouProbablyKnow commented:
> Just if the person spends some time learning the job on a lower-competency job first, then there will be lower chances he will try to abuse his powers. Lower but not absolute chance.

Why? They'll start as cadet and then graduate to officer, while all the way still being exposed to shitsec behavior. I really don't understand how that's gonna make them less likely to abuse their power. 

Maybe at first while they have less of it to abuse but as you said the hours total is the same

DrDiasyl commented:
> > Just if the person spends some time learning the job on a lower-competency job first, then there will be lower chances he will try to abuse his powers. Lower but not absolute chance.
> 
> Why? They'll start as cadet and then graduate to officer, while all the way still being exposed to shitsec behavior. I really don't understand how that's gonna make them less likely to abuse their power. 
> 
> Maybe at first while they have less of it to abuse but as you said the hours total is the same

This is where power of code stops, we cannot using code make it so they're being nurtured only by good seccies. But we can only hope so, and I believe into it.

DrDiasyl commented:
> Im worried itl make the job screen alittle too cluttered and complex for brand new players and the names for some of the jobs dont do much to help (mainly test subject)
> 
> TLDR: Assistant inflation.

What isn't the funniest way to learn science then being directly in it? Of course nothing else!
If they're willing to learn, they totally will get how science department functions.

SomeoneYouProbablyKnow commented:
Strictly speaking you can do more withcode, change the incentives yada yada but yeah that's way out of scope for this PR. I don't even dislike it, just saying that shitsec will be unaffected by this change

DrDiasyl commented:
> I don't think I like this at all.
> 
> 1. I think incompetent crew members in important positions is good for the game and creates more interesting stories than every doctor/engineer knowing how to do everything. Of course after one round on this you won't be a master, but I think the implied values of this PR are misguided. For example, I completely disagree that "Decrease the number of shitsecs by making the learning curve for Security more logical" is a good thing--you and I do not see the game in the same way.
> 2. If you're not giving them the full gear and access, you're not really teaching them the role, since when they get the "full" role, they're still going to have another, different learning curve. People in these roles are going to have to do more that the role they actually want to play does.
> 3. Opening more job slots shouldn't be treated as an innate good. We have them for a reason and increasing them should be done carefully. Something like increasing job slots of security (assuming you haven't matched the security config, which you might have) for instance is definitely a pretty large change.
> 4. It makes the jobs menu more cluttered with jobs experienced players shouldn't actually really be playing.
> 
> I think what I would want more is the ability for intern stuff to apply to specific jobs. For instance, less than 2 hours as the job slot prefixes "Junior" to your ID card.
> 
> Fikou wants a test merge and I am okay with that but I don't want this merged

Also, you said you want for that "Intern" system to be expanded. If I will PR something like TGMC/CM rank system then it will be fine?

Like if you only joined security officer you're "Private Security Officer" or "Security Cadet" and more you play you become "Security Officer" and then maybe for playing really alot get fancy title like "Security Sergeant".

Those won't obviously grant any authority over the others, but just show off how noob/pro you're.

Wallemations commented:
Not digging into the more nuanced parts of the PR, I'm wondering why bother making a Nurse job when you say that the paramedic is equivalent to their "starter job"?

DrDiasyl commented:
> Not digging into the more nuanced parts of the PR, I'm wondering why bother making a Nurse job when you say that the paramedic is equivalent to their "starter job"?

Paramedic doesn't do the same things a Medical Doctor or Nurse does.
Paramedic runs around station retrieving dying people.

While Medical Doctor and Nurse spent their entire time in the medical department.

EricZilla commented:
I think people are overestimating the amount of audacity the average person has. Speaking from my own experience, I haven't even tried doing real medical because of how complex the pharma side is, and generally just go in as an extra helping hand as psych, so having a job where I'm supposed to be bossed around by the other medical people would be a really great way to get a foot in the door. 
Engineering really doesn't need this as much, because as soon as you do one shift you'll realize how simple it is, but this would also greatly increase the amount of people who put their foot in the door.

That said, allowing some sort of time gate title unlock would probably be the best way to do that, and add a good way to tell at a glance if you are dealing with a schmuck or a pro.

Test subject is amazing and should be added all on its own. You'd be fulfilling every science main's mad scientist fantasies by giving them a little igor ass motherfucker to electrocute. Honestly even if its not added, I'm going to start requisitioning prisoners for this when possible lmao.

I also think a low ranking bar role like waiter would be nice just for having someone who isnt busy who can help the rest of service when they need someone. Theoretically assistants can do that, but we all know they aren't going to.

I think my perfect world version of all this would be adding the titles and an alternate outfit to low ranking people in a job, then maybe adding a upper tier too.

Example:
Nurse - Scrubs
Doctor - Normal doctor shit
Senior Practitioner - They get a cane and a brown lol house md headass

Then maybe add an option to choose which one you spawn in as part of the job selection in character settings.

Mothblocks commented:
@Fikou are u getting a key holder to tm this

DrDiasyl commented:
> @Fikou are u getting a key holder to tm this

Its currently test merged at the Manuel

Wallemations commented:
Part of me feels like these should be some sort of preset for assistants rather than their own proper "jobs", for example think of how you can set your preferred department as a security guard.

I'll admit I don't know all the brass tacks related to the change you're trying to make here so take this suggestion lightly I guess.

xPokee commented:
I'm really not a fan of this at all. When someone that has almost a 2000 hour playtime and has been playing sec recently (but not enough playtime to unlock officer) has to now play the beginner roles. Yeah, this is stupid.

DrDiasyl commented:
> I'm really not a fan of this at all. When someone that has almost a 2000 hour playtime and has been playing sec recently (but not enough playtime to unlock officer) has to now play the beginner roles. Yeah, this is stupid.

1. Press F1
2. Enter "I have 2000 hours in this game, unlock all jobs"
3. You get all jobs unlocked.

YehnBeep commented:
I just kind of wanted to add a note on why I think this could be a good contribution, though others have suggested alternate ways to do it - and maybe those could be an improvement and there can be general refinements - but I think the overall concept is good.

Speaking as someone that has played an assistant that actually does try to help departments, there's basically 2 problems.

1. Your accesses can end up really weird. Like, for instance, if you wanted to help out medical - you have chem access but not medical department access, and maybe what you needed was pharmacy access. I've seen this sort of thing from both HoPs and department heads - not many people seem to understand accesses very well.
2. A lot of people get very defensive when an assistant is in their department (even explicitly authorized) - and just flat out don't want you there. This can be somewhat mitigated if you're well known, but even then.

Now, that said, 15 hours may be a bit much for some. That's 8 rounds if they all go for a full 2 hours (So, in practice, more). Maybe for security that's OK, but, for basic jobs in other departments, that seems a pretty steep requirement to have before they can learn it "for real" with all the tools and such it entails.

Also bearing in mind that with the current setup, it's not 15hrs overall experience with the game. It's 15 hours on that particular job's department, starting over again each time you switch departments.

Thank you for considering.

DrDiasyl commented:
> I just kind of wanted to add a note on why I think this could be a good contribution, though others have suggested alternate ways to do it - and maybe those could be an improvement and there can be general refinements - but I think the overall concept is good.
> 
> Speaking as someone that has played an assistant that actually does try to help departments, there's basically 2 problems.
> 
> 1. Your accesses can end up really weird. Like, for instance, if you wanted to help out medical - you have chem access but not medical department access, and maybe what you needed was pharmacy access. I've seen this sort of thing from both HoPs and department heads - not many people seem to understand accesses very well.
> 2. A lot of people get very defensive when an assistant is in their department (even explicitly authorized) - and just flat out don't want you there. This can be somewhat mitigated if you're well known, but even then.
> 
> And I think maybe this addition could help a lot with this.
> 
> Now, that said, 15 hours may be a bit much for some. That's 8 rounds if they all go for a full 2 hours (So, in practice, more). Maybe for security that's OK, but, for basic jobs in other departments, that seems a pretty steep requirement to have before they can learn it "for real" with all the tools and such it entails.
> 
> Also bearing in mind that with the current setup, it's not 15hrs overall experience with the game. It's 15 hours on that particular job's department, starting over again each time you switch departments.
> 
> Thank you for considering.

Only sec gets their previous 300 minutes split between crew and Cadet.
Other jobs require only 60 minutes.

YehnBeep commented:
> Only sec gets their previous 300 minutes split between crew and Cadet. Other jobs require only 60 minutes.

Is this a very recent change? On the test merge it was showing me 15 hours (I can share a screenshot if needed).

DrDiasyl commented:
Okay, from those 3 rounds that passed since it got test merged I can make out a few things:
- People love the Test Subject, science really sparked up after getting their own personal D-Class personnel (and also AI trying to advocate "safe experiments" to prevent human test subject's harm)
- Not alot of people said anything about Nurse or Maintenance Technician, but in general their reception was *fine.* Although I seen a interesting sighting, Maintenance Technician's seem to actually maintain and fix station more then actual engineers! Really interesting.
- Cadets only work when the security department is well staffed. When security department is badly staffed Security Cadets have to be converted just into Security Officers. As security really needs fighting force, but they got 3 unarmed people without security lockers access. But in well staffed departments they fill out jobs like manning posts around station, and also helping officer's.

Gr1lledcheese commented:
![image](https://user-images.githubusercontent.com/45194738/230747556-8f25904d-cdf8-4cda-be69-f96ff6158c5a.png)

Pls fix
I have about 2 hours on robo and 3 hours on scientist, could you make having an hour or so logged in a department job disable the timelock because as of right now it's a pain in the ass. Not opposed to the idea once you get it polished :)

Cheshify commented:
security is rolling antags by the way.

Fikou commented:
its untestmerged now its been fun

DrDiasyl commented:
> ![image](https://user-images.githubusercontent.com/45194738/230747556-8f25904d-cdf8-4cda-be69-f96ff6158c5a.png)
> 
> Pls fix
> I have about 2 hours on robo and 3 hours on scientist, could you make having an hour or so logged in a department job disable the timelock because as of right now it's a pain in the ass. Not opposed to the idea once you get it polished :)

@YehnBeep really interesting, maybe those can be customised in the configs perhaps? Because in the code its set to 60 minutes only.

YehnBeep commented:
> > ![image](https://user-images.githubusercontent.com/45194738/230747556-8f25904d-cdf8-4cda-be69-f96ff6158c5a.png)
> > Pls fix
> > I have about 2 hours on robo and 3 hours on scientist, could you make having an hour or so logged in a department job disable the timelock because as of right now it's a pain in the ass. Not opposed to the idea once you get it polished :)
> 
> @YehnBeep really interesting, maybe those can be customised in the configs perhaps? Because in the code its set to 60 minutes only.

I'm not sure. My experience with it was on the PR on TG's Manuel server, so I'm not sure what configuration options there might have been available or in play - just that it seemed everything affected was subject to the 15hr timelock.
